contracts/secret-storage/TeeManager.sol

```diff
@@ -554,7 +558,19 @@ contract TeeManager is
     ) internal {
         uint256 stakeAmount = teeNodes[_enclaveAddress].stakeAmount;
         // compounding slashing formula: remainingStakeAmount = stakeAmount * (1 - (r/100)) ^ n
-        uint256 remainingStakeAmount = stakeAmount * ((SLASH_MAX_BIPS - SLASH_PERCENT_IN_BIPS) ** _missedEpochsCount) / (SLASH_MAX_BIPS ** _missedEpochsCount);
+        // uint256 remainingStakeAmount = stakeAmount * ((SLASH_MAX_BIPS - SLASH_PERCENT_IN_BIPS) ** _missedEpochsCount) / (SLASH_MAX_BIPS ** _missedEpochsCount);
+        
+        uint256 remainingStakeAmount = stakeAmount;
+        uint256 iterations = _missedEpochsCount / 12;
+        for (uint256 i = 0; i < iterations; i++) {
+            remainingStakeAmount = remainingStakeAmount * ((SLASH_MAX_BIPS - SLASH_PERCENT_IN_BIPS) ** 12) / (SLASH_MAX_BIPS ** 12);
```

Comment:
I think * operator can also overflow because of value of remainingStakeAmount, it's in order of 10^18

Reply:
Math library by openzeppelin provides the muldiv function, that can do this operation with highest precision. Will be using that.